src/chatdlg.cpp

```diff
@@ -106,8 +106,26 @@ void CChatDlg::OnClearChatHistory()
 void CChatDlg::AddChatText ( QString strChatText )
 {
     // add new text in chat window
-    txvChatWindow->append ( strChatText );
-
-    // notify accessibility plugin that text has changed
-    QAccessible::updateAccessibility ( new QAccessibleValueChangeEvent ( txvChatWindow, strChatText ) );
+	
+    // analyze strChatText to check if hyperlink
+	if (strChatText.contains("https://", Qt::CaseInsensitive) | strChatText.contains("http://", Qt::CaseInsensitive)) 
+	{
+		int indx_http  = strChatText.indexOf("http", 0);
```

Comment:
So if I say "the http protocol is indicated by a 'http://' or 'https://' prefix on a url", this will find the first "http" (between "the" and "protocol").

Reply:
That's correct. 
A better way would be to find the index of strChatText.indexOf("http://", 0) or strChatText.indexOf("https://", 0) and include some more checks.
But in any case if the link appears to be invalid the browser will deal with it and depict it as non sense.

Reply:
Firefox opens (from your post above) both equally well.  If it's a defect in the QTextBrowser widget's URL handling, that's a different matter.  (And another reason to take care with using it for active links.)

Reply:
I don't think relying on the browser to clean up is a safe option.

Reply:
Hi Peter
IMHO The browser will merely not open the wrong url and issue a 404 page.

Reply:
Take a look at Qt's [QUrl](https://doc.qt.io/qt-5/qurl.html).  A simple check like `QUrl(text).isValid()` would go a long way.